lib/ball.js

```diff
@@ -23,50 +23,57 @@ let ball = class Ball {
     this.x += this.xSpeed;
     this.y += this.ySpeed;
     return this;
-  };
+  }
 
   wallCollision(canvas) {
-    //left or right
-    if(this.x + this.xSpeed > canvas.width - this.radius ||
-       this.x + this.xSpeed < 0 + this.radius) {
+    //left
+    if(this.x + this.xSpeed < this.radius) {
+      this.x = this.radius;
```

Comment:
Here (and lines 36, 41, & 46), it now sets the ball on the canvas edge before reversing the momentum. This just makes sure that it can't get shoved into the wall by the paddle-grab "feature"